plugins/outputs/plugin.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Aspect Build Systems, Inc.
+ *
+ * Licensed under the aspect.build Commercial License (the "License");
+ * you may not use this file except in compliance with the License.
+ * Full License text is in the LICENSE file included in the root of this repository.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	goplugin "github.com/hashicorp/go-plugin"
+
+	"aspect.build/cli/pkg/bazel"
+	"aspect.build/cli/pkg/plugin/sdk/v1alpha2/config"
+	aspectplugin "aspect.build/cli/pkg/plugin/sdk/v1alpha2/plugin"
+)
+
+func main() {
+	goplugin.Serve(config.NewConfigFor(New()))
+}
+
+type Outputs struct {
+	aspectplugin.Base
+}
+
+func New() *Outputs {
+	return &Outputs{}
+}
+
+func (plugin *Outputs) CustomCommands() ([]*aspectplugin.Command, error) {
+	return []*aspectplugin.Command{
+		aspectplugin.NewCommand(
+			"outputs",
+			"Inspect declared outputs.",
+			"Queries for the outputs declared by actions generated by the given target.",
+			func(ctx context.Context, args []string, bzl bazel.Bazel) error {
```

Comment:
It would be beneficial to have a top-level function that returns this and allows injecting some dependencies for testing (if needed, otherwise this plugin will just be a good reference for future plugins). Also will help with reducing a bit the indentation of the whole implementation.

```go
func NewOutputsCommand() aspectplugin.CustomCommandFn {
    return func(ctx context.Context, args []string, bzl bazel.Bazel) error {
        ...
    }
}
```

Reply:
I think I understand this, just means "extract function with a fancy Factory signature"